### ISSUE TITLE
News and Posts subpage – implementation and preparation

### ISSUE THREAD

**Issue by linnutee** [decred/dcrweb#561]
Preview: https://xd.adobe.com/view/89fa7693-12b8-43a0-4841-32a12f827c8b-ffcb/

Now that we got the hugo implementation, got this long in the pipeline item addressed. 

Front-end implementation assigned for @rmykolas  – design source files, specs, assets: https://github.com/decred/dcrdesign/issues/8

- As previously planned, the "news" subpage will replace the temporary Press subpage. 
- All posts can be either external links (displaying url and icon hover – nb! update the icon, i've increased the contrast slightly) or content post pages. 
- Most of the design is iterated off the press subpage. Have added a few more design patterns for posts: possibility for hero images, in content images and captions. More styling patterns can be added as follow-up on need basis. 
- Posts will need to open in separate pages to keep page load optimal. 
- Category selection has a slightly updated selector pattern, which i'll make a separate issue for updating in the contributors page also.
- As a follow-up update, can introduce "pinned posts". This should replace #423 and http://decred.org/recruiting/.
- Initial categories are the following, based on content going through twitter and feedback received (these can be updated on need basis): 

```
**Announcements**, containing following post types:
- Announcements (e.g. old dcr blogposts, and ideally replacing the future ones)
- Release Updates
- Service announcements (saved for any rare emergency notices)

**Research and Development**, containing following post types:
- General Research and Writings 
- Dev Posts/Mentions/Research

**Community**, containing following post types:
- General community posts (e.g. noteworthy discussions, mentions)
- Show DCR (e.g. user projects and initiatives)
- Broadcast, streams, videos

**Media and Events**, containing following post types:
- Decred in media (half of the current press page) 
- Press releases, interviews (other half of the current press page) 
- Events and Meetups

**Contributing**, containing following post types:
- Help needed (e.g. norteworthy  gh issues)
- RFPs

Noting that the post types are only descriptive and use icons 
for their representation for now. How well these function and 
get populated can be seen after some usage and optimized then. 
```

**Naming feedback/discussion**
Interested in getting some feedback regarding the naming of this subpage as only "News" isn't most accurate. The page is a type of portal that is both a publishing source of announcements and releases as well a recollection of outside mentions, contribution opportunities, posts, events etc. Did consider names such as "Portal" "Newsfeed" "News and Feeds", "Newsportal", "Newsboard" "Broadcast" "Comms" and such … so far "News and Posts" seems most accurate and rolls off the tongue. Better options what are more accurate and not obscure welcomed. 

**Content for launch / help needed**
Ideally the subpage should be launched with pre-populated content for all categories. For example all of the announcements from blog.decred.org should be linked, perhaps some of the recent or bigger dev dispatches, recent and more mention worthy research bits. If anyone can pick up this work, best to make a separate issue for planning the content, thanks.

**Comment by peterzen** [comment]
`> * Posts will need to open in separate pages to keep page load optimal.`

For content items such as Press releases, which are internal links, the opening of a new window may be unnecessary.  Instead of the current JS based accordion layout it would be best to render the news items on separate pages (under unique URLs) so that this content becomes accessible to search engines, as well.

**Comment by linnutee** [comment]
Yup that's what I meant. Staying in the same browser window but different pages instead of everything in accordions.

**Comment by Dustorf** [comment]
Exciting initiative. I'm supportive of the idea but think we can condense the buckets from 5 to max 3. What about:
News & Events (would include announcements)
Original Content (research, writing)
Help Wanted (FRPs, issues)

I also agree we need a good way to describe it collectively. The first two go together easily and could be described by your suggestions. What if we move the RFPs/issues over to the new Self-Funding subpage? It would describe the contractor model, then list the available ways to work for Decred

**Comment by oregonisaac** [comment]
```
Would it be possible to move on the "meetup" portion?  Seeing that as a stronger push I would like to be able to advertise the meetups happening through Telegram and other social media through an official channel like the website so folks from around the world who have connected virtually can start meeting in person.
Some meetups I know about:
Vancouver BC, CA - https://www.meetup.com/Decred/
Toronto, ON, CA - https://www.meetup.com/Decred-GTA/
Portland OR, USA - https://www.meetup.com/decredpdx/
Melbourne, Australia - https://www.meetup.com/Decred-Australia/
Katowice, Poland - https://www.meetup.com/Decred-Polska/

There are probably more I don't know about, and it would be great to have them promoted.  Perhaps @Dustorf has a more extensive list.
```

**Comment by Dustorf** [comment]
People are acting pretty autonomously. There will be a SF meetup on March 6th, but there aren't more I know about. It will be great to allow people to share info with the web team to get them on the site easily in the future

**Comment by peterzen** [comment]
> ```
> Would it be possible to move on the "meetup" portion?  Seeing that as a stronger push I would like to be able to advertise the meetups happening through Telegram and other social media through an official channel like the website so folks from around the world who have connected virtually can start meeting in person.
> Some meetups I know about:
> Vancouver BC, CA - https://www.meetup.com/Decred/
> Toronto, ON, CA - https://www.meetup.com/Decred-GTA/
> Portland OR, USA - https://www.meetup.com/decredpdx/
> Melbourne, Australia - https://www.meetup.com/Decred-Australia/
> Katowice, Poland - https://www.meetup.com/Decred-Polska/
> 
> There are probably more I don't know about, and it would be great to have them promoted.  Perhaps @Dustorf has a more extensive list.
> ```



> ```
> Would it be possible to move on the "meetup" portion?  Seeing that as a stronger push I would like to be able to advertise the meetups happening through Telegram and other social media through an official channel like the website so folks from around the world who have connected virtually can start meeting in person.
> Some meetups I know about:
> Vancouver BC, CA - https://www.meetup.com/Decred/
> Toronto, ON, CA - https://www.meetup.com/Decred-GTA/
> Portland OR, USA - https://www.meetup.com/decredpdx/
> Melbourne, Australia - https://www.meetup.com/Decred-Australia/
> Katowice, Poland - https://www.meetup.com/Decred-Polska/
> 
> There are probably more I don't know about, and it would be great to have them promoted.  Perhaps @Dustorf has a more extensive list.
> ```

+1 for such an event calendar section, it would be very useful.

**Comment by xaur** [comment]
> Interested in getting some feedback regarding the naming of this subpage as only "News" isn't most accurate. The page is a type of portal that is both a publishing source of announcements and releases as well a recollection of outside mentions, contribution opportunities, posts, events etc. Did consider names such as "Portal" "Newsfeed" "News and Feeds", "Newsportal", "Newsboard" "Broadcast" "Comms" and such … so far "News and Posts" seems most accurate and rolls off the tongue.

@linnutee thinking from "subscriber" perspective, we have a ton of possible updates that fall into different categories. Each category should have its URL path and a subscription feed (e.g. RSS). For example, `/media`, `/contribute` or the long needed `/events`.

A special category can aggregate all updates, e.g. "All News", but I'd avoid using or inventing abstractions like "Portal" or "Broadcast" as the are too vague and have a corporate feel.

**Comment by linnutee** [comment]
Re: Xaur
The naming case applies for the subpage as a whole and not so much around categories. Brought it up as “News” or those other options aren't the most accurate terms to describe this aggregate-type subpage, as this is both a feed and publication. 

Good point on the subscribing approach, I guess this would be a development task to get sorted. Not very aware on what's the best practice on these feeds since RSS has been somewhat killed off. Certainly would be a plus to have the structure set so that all content could be pulled to Feedly, Newsblur and any other popular services.

Re: Dustin: 
- Adding RFPs/issues into another subpage will create a conflict in the user flow. This is because the “news and posts” subpage takes a function of a repeatedly visited page (that receives new content on regular basis), whereas static content subpages are likely visited only once or after long periods of time. 
- Three categories won’t be able to house the content properly.   Basically Announcements is the purest category and in top of the hierarchy. On suggestion "News and Events" – this will dialute the importance of Announcements (e.g. dcrblog posts) as it would be unfitting to have posts like "Release Update XYZ" or "Politeia Release" next to “Thursday evening #trading beer meetup”
- I do agree Events perhaps being most out of context and not ideal to lump them in with the "Media" (being basically current press page and likely with largest quantity of common entries). Though in order just to get this finally out (being in the pipeline behind this CMS case for over a year), I wouldn't really grind over this too much and leave either the events as a separate category or add them under the Community category instead. As a future update, a separate solution for events would be ideal, but this will be also a somewhat large task to do properly – can think of benchmarks like https://www.ted.com/tedx/events
https://www.eventbrite.com/d/estonia--tallinn/events/
that would allow for bit more specific filtering – location and time based and also a calendar view would be super helpful.
-  Research and Dev vs Community categories – these two categories mainly differ on the type of content, whilst “Community” contains a wide array of content types, from shows, podscasts etc – I find it wise to have more technical, academic pieces grouped together under Research/Dev.

Based on feedback, either simple approach on category and post breakdowns could go: 

Categories/Posts A
```
**Announcements**, containing following post types:
- Announcements (e.g. old dcr blogposts, and ideally replacing the future ones)
- Release Updates
- Service announcements (saved for any rare emergency notices)

**Research and Development**, containing following post types:
- General Research and Writings 
- Dev Posts/Mentions/Research

**Community**, containing following post types:
- Show DCR (e.g.  general posts, noteworthy discussions, mentions, user projects etc)
- Broadcast, streams, videos (e.g. DA, Decred in Depth, Roundtable etc)
- Events and Meetups

**Media**, containing following post types:
- Decred in media (half of the current press page) 
- Press releases, interviews (other half of the current press page) 

**Contributing**, containing following post types:
- Help needed (e.g. norteworthy  gh issues)
- RFPs
```

Categories/Posts B
```
**Announcements**, containing following post types:
- Announcements (e.g. old dcr blogposts, and ideally replacing the future ones)
- Release Updates
- Service announcements (saved for any rare emergency notices)

**Research and Development**, containing following post types:
- General Research and Writings 
- Dev Posts/Mentions/Research

**Community**, containing following post types:
- Show DCR (e.g.  general posts, noteworthy discussions, mentions, user projects etc)
- Broadcast, streams, videos (e.g. DA, Decred in Depth, Roundtable etc)

**Events**, containing following post types:
- Community Meetups
- Decred in Conferences

**Media**, containing following post types:
- Decred in media (half of the current press page) 
- Press releases, interviews (other half of the current press page) 

**Contributing**, containing following post types:
- Help needed (e.g. norteworthy  gh issues)
- RFPs

```

**Comment by Dustorf** [comment]
I think there's a lot of value making the events obvious, so my preference is B

**Comment by peterzen** [comment]
In the Events section, it would be awesome to have an iCal feed as well, so that people can subscribe directly in their calendar.

**Comment by xaur** [comment]
Is there an issue to discuss events in dcrweb specifically? It will require an active maintainer and some nice YAML structure that would allow to build nice filtering features later.

For more general needs I made [shared calendar](https://github.com/xaur/decred-issues/issues/116) issue.

**Comment by xaur** [comment]
Thinking about content management now.

It's great to have the data from which the site is built in separate nicely structured and versioned files. Git allows to replicate all collected knowledge to multiple people and keep it in sync.

When all these new pages go live and people start to actively add content, we will see a lot of commit churn around "site content" files (md pages, yaml data, non-layout images), and it will be harder to dig changes for "site code" files (css, js, layouts). Effectively, streams of commits from website programmers and content managers would overlap all the time as it happens today, but in greater scale.

It might be useful to separate "content" files into a separate Git repo or an orphan branch.

**Comment by peterzen** [comment]
The theme code itself doesn't change that often and in the current Hugo structure the theme is reasonably separated from the content - unless there is a major refactoring going on at the same time with heavy content changes overlaps should be non-existent to minimal.

**Comment by peterzen** [comment]
I've implemented the sections/content types in Hugo, according to the B) site map above. It's in a separate repo for now as there are some questions/clarifications that the spec doesn't cover need to be iterated.  It would also be good to add more initial content whilst the details are worked on in terms of content types/navigation etc.

https://dcrweb.netlify.com/news/

### Questions

- Second level navigation - the [Media](https://dcrweb.netlify.com/news/media/) section really calls for a second level nav as the best practice would be to keep the *Decred in the Media* and *Press releases* subsections separate.  We may want to do the same in the *Announcements*?

- Any thoughts/comments on the [Events](https://dcrweb.netlify.com/news/events/) page?  What type of content would be published here.  I can see that it would be nice to pull in calendar feeds from meetup groups to provide a comprehensive view on what's on in the Decred world.

- Example content is needed in [Contributing](https://dcrweb.netlify.com/news/contributing/), [Community](https://dcrweb.netlify.com/news/community/) - @Dustorf is this perhaps something you could contribute?  

 - Any other comments on structure/navigation?

**Comment by linnutee** [comment]
Cool, very nice to see this moving on.

Answering 1st and 2nd bulletpoints.
- I would ideally avoid having two level navigation. At least from the initial launch until there's actually need due to the amount of content. This way the structure remains simpler and can be grown more accurately based on need. So if anything the 1st level category structure could become a little bit more elaborate. But that's up to the contributors essentially. 

Also instead of subcategories, there are icon based tags to help create order inside each category. 

- Calendar: in the present design it would be posts linked to the event source. Ie. meetup.com, event sites etc. Calendar view would certainly be a nice enhancement. Unless there's any significant issues with building it, ideally the calendar can be another "viewing mode". Meaning there's an option to switch between the normal post list + calendar view. Reason being as this way it would be a little bit more integrated to the subpage. Lmk and can get this in the pipeline to get the design thought through and created.

**Comment by peterzen** [comment]
> * I would ideally avoid having two level navigation. At least from the initial launch until there's actually need due to the amount of content. This way the structure remains simpler and can be grown more accurately based on need. So if anything the 1st level category structure could become a little bit more elaborate. But that's up to the contributors essentially. Also instead of subcategories, there are icon based tags to help create order inside each category.

I kind of agree that it's not very fortunate to add another nav level - however, it is not a good idea to lump the external links and the press releases together. In the interest of having a clear separation between 'syndicated' content and official communication I think we need to break up the Media page in some way - maybe keep *Decred in the media* and *Press releases* on separate top-level pages?

> * Calendar: in the present design it would be posts linked to the event source. Ie. meetup.com, event sites etc. Calendar view would certainly be a nice enhancement. 

Agreed.  Also, I think a location based view would also be valuable for people looking for IRL events. 

> Unless there's any significant issues with building it, ideally the calendar can be another "viewing mode". Meaning there's an option to switch between the normal post list + calendar view. Reason being as this way it would be a little bit more integrated to the subpage. Lmk and can get this in the pipeline to get the design thought through and created.

Yes please.

**Comment by Dustorf** [comment]
Thanks, Peter. Agree with both of you on the first points:
1. keep things as simple as possible, particularly to start
2. I don't think there is a need to have a calendar view. Simply listing the events in time order should be sufficient. These events include: local meetups, conferences Decred attends, AMAs, livestreams.
*Community-this is for links to newer content. So, when Decred in Depth podcast of a Decred Assembly episode goes live, we can add a link here. 
*For Research & Dev., @RichardRed0x is working up a website to showcase much of the work, so it probably makes sense to simply list links here.

Aggregating info here will streamline production of the Decred Journal. LMK if I missed anything or if you have further ?'s.

**Comment by peterzen** [comment]
@linnutee Could you please export the news section icons into SVG files?  I can't seem to find the vector icons in the XD file.

**Comment by linnutee** [comment]
@peterzen added here: [dcrweb - news subpage - icons.zip](https://github.com/decred/dcrweb/files/3772880/dcrweb.-.news.subpage.-.icons.zip)

**Comment by peterzen** [comment]
Thank you @linnutee

**Comment by Dustorf** [comment]
At this time, I think we can streamline it down to the following:
* Announcements 
* Research
* Media & Events
dev posts fall under announcements, there are a number of contributors @RichardRed0x Checkmatey and Permabull who are or have built out their own research websites. Community isn't worth it, and anything notable could fall under Media & Events. Contributing is better off hosted by individuals on GH

**Comment by xaur** [comment]
Will Research host full copies of research reports or have short summaries and link to full version elsewhere?

**Comment by Dustorf** [comment]
I think it should simply link to the eternal links to keep things simple

**Comment by peterzen** [comment]
Is blog.decred.org going to be maintained seperately, as currently?  Or become part of www.decred.org (I think this was discussed before, maybe @behindtext has a preference?)

**Comment by xaur** [comment]
More and more I like the idea to split textual content into a separate repo that can be "plugged into" the static site repository (with templates, css, js, assets) as a git submodule.

Earlier I [noted](https://github.com/decred/dcrweb/issues/561#issuecomment-471165915) the overlapping churn of commits as a motivation, but a bigger reason I see now is to allow for quick replication and sync of text content, even over a slow link. dcrweb is almost 200 MB now, mostly due to heavy assets while the text part is <1 MB. Text content is the most interesting to preserve in context of censorship resistance.

Another bonus is that copywriters can clone the tiny text repo and work with it without getting the full 200 MB thing.

Multiple such text repos like "website" and "blog" can be plugged into the site code repo as submodules and rendered with the single codebase of templates, styles and other assets.

**Comment by jholdstock** [comment]
I've reviewed this whole issue and I think we have everything pretty much covered. Closing.
